File: src/main/java/com/elchologamer/userlogin/listener/restriction/ItemDropRestriction.java
package com.elchologamer.userlogin.listener.restriction;

import org.bukkit.event.EventHandler;
import org.bukkit.event.player.PlayerDropItemEvent;

public class ItemDropRestriction extends BaseRestriction<PlayerDropItemEvent> {
    public ItemDropRestriction() {
        super("itemDrop");
    }

    @EventHandler
    public void handle(PlayerDropItemEvent e) {
        if (shouldRestrict(e)) e.setCancelled(true);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/restriction/ItemDropRestriction.kt
package com.elchologamer.userlogin.listener.restriction

import org.bukkit.event.EventHandler
import org.bukkit.event.player.PlayerDropItemEvent

object ItemDropRestriction : BaseRestriction<PlayerDropItemEvent>("itemDrop") {
    @EventHandler
    fun handle(e: PlayerDropItemEvent) {
        if (shouldRestrict(e)) e.isCancelled = true
    }
}